arcaflow_plugin_rtla/rtla_plugin.py

```diff
@@ -121,12 +123,103 @@ def run_timerlat(
         if self.finished_early:
             timerlat_proc.send_signal(2)
 
-        if params.enable_time_series and trace_file_exists:
+        # Example time series output from the tracer (truncated)
+        # <idle>-0      [009] d. 123.769498: #1002 context    irq timer_latency  458 ns
+        #  <...>-625890 [009] .. 123.769499: #1002 context thread timer_latency 666 ns
+        #  <...>-625890 [009] .. 123.769500: #1002 context user-ret timer_latency 499 ns
+        # <idle>-0      [002] d. 123.769528: #1003 context    irq timer_latency  587 ns
+        #  <...>-625883 [002] .. 123.769532: #1003 context thread timer_latency 712 ns
+        #  <...>-625883 [002] .. 123.769534: #1003 context user-ret timer_latency 462 ns
+
+        if params.enable_time_series and trace_file:
             # Interrupt the time series collection process and capture the output
             timeseries_proc.send_signal(2)
-            timeseries_output, _ = timeseries_proc.communicate()
+            timeseries_file.close()
+            trace_file.close()
+
+            # The calculation of the offset from uptime to current time certainly means
+            # that our time series is not 100% accurately aligned to the nanosecond, but
+            # the relative times will be accurate. We'll accept this as good enough.
+            uptime_offset = time.time() - time.monotonic()
+
+            with open("./timeseries_file") as timeseries_output:
+
+                if all(False for _ in timeseries_output):
+                    print(
+                        "No results reading tracer output; "
+                        "Skipping time series collection\n"
+                    )
```

Comment:
I'm not following how you are seeing that the first line gets eaten.

```
>>> i = ["a", "b", "c", "d"]
>>> if all(False for _ in i):
...   print("bad juju")
... 
>>> for line in i:
...   print(line)
... 
a
b
c
d
```

```
>>> i = []
>>> if all(False for _ in i):
...   print("bad juju")
... 
bad juju
>>> for line in i:
...   print(line)
... 
>>> 
```

Reply:
Try your experiment with `i` set to an iterator or a generator.  With it set to a `list`, of course you see the first item when you check after the `all()` call, because the _list_ state is **unchanged**.  😁